src/main/python/hydra/lib/hydrabase.py

```diff
@@ -77,17 +79,45 @@ def _execute(self, method, **kwargs):
             func(**kwargs)
             ha.stop()
 
-    def _get_group_info(self):
-        """
-        Return group info
-        """
-        return self.group_info
+    def add_tasks_to_group(self, tasks_list):
+        """
+        Add tasks to the group.
+        :param tasks_list: Tasks list to add. Duplicate tasks will be added only once.
+        :return: 0 in case of success.
+        """
+        in_first = set(self.tasks_list)
+        in_second = set(tasks_list)
+        in_second_but_not_in_first = in_second - in_first
+        self.tasks_list.extend(in_second_but_not_in_first)
+        return 0
+
+    def delete_tasks_from_group(self, num_tasks_to_delete):
+        """
+        Delete given number of tasks from group.
+        :param num_tasks_to_delete: Number of tasks to be deleted from group.
+        :return: 0 in case of success. 1 in case of failure.
+        """
+        res = 0
+        for i in range(num_tasks_to_delete):
+            try:
+                task_id_to_del = self.tasks_list.pop().rsplit('_', 1)[0]
+                self.hydra.mt.kill_task(self.app_name, task_id_to_del)
```

Comment:
previously, a group was just a collection of ip, port info at a higher level. Now deleting from group means we are actually killing the task. This info should be propagated to the user, in comments of the function and with some logs.